### PR TITLE
Remove CosmWasm functionality from BCP

### PR DESCRIPTION
**Pull request by webmaster128**
~Based on #183~

This is step 2 from #179

- [x] Make bcp package independent of cosmwasm
- [x] Rename CosmWasm* -> Cosmos*

### CODE REVIEW

**Comment by ethanfrey** [comment]
I will review this when #179 is done so the diff is manageable

**Comment by webmaster128** [comment]
@ethanfrey ping

File: packages/bcp/src/cosmwasmcodec.spec.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import { Address, PostableBytes, PrehashType, SendTransaction, TokenTicker } from "@iov/bcp";
+import { PostableBytes, PrehashType } from "@iov/bcp";
 import { Encoding } from "@iov/encoding";
 
-import { CosmWasmCodec } from "./cosmwasmcodec";
+import { CosmosCodec } from "./cosmwasmcodec";
```
Comment (ethanfrey) [inline packages/bcp/src/cosmwasmcodec.spec.ts:4]: why is the file still named `cosmwasmcodec` if it is now a more generic `CosmosCodec`?

File: packages/bcp/src/cosmwasmconnection.ts [hunk 2]
```diff
@@ -85,50 +82,47 @@ function compareByHeightAndHash(a: IndexedTx, b: IndexedTx): number {
 /** Account and undefined are valid events. The third option means no event fired yet */
 type LastWatchAccountEvent = Account | undefined | "no_event_fired_yet";
 
-export class CosmWasmConnection implements BlockchainConnection {
+export class CosmosConnection implements BlockchainConnection {
```
Comment (ethanfrey) [inline packages/bcp/src/cosmwasmconnection.ts:85]: Rename and simplification is good.

I would rename the file as well.

File: packages/bcp/src/decode.spec.ts [hunk 3]
```diff
@@ -159,40 +140,7 @@ describe("decode", () => {
           ],
         },
       };
-      expect(parseMsg(msg, defaultMemo, testdata.chainId, defaultTokens, defaultErc20Tokens)).toEqual(
-        defaultSendTransaction,
-      );
-    });
-
-    it("works for ERC20 send transaction", () => {
```
Comment (ethanfrey) [inline packages/bcp/src/decode.spec.ts:167]: that's easier

File: packages/bcp/src/cosmwasmcodec.spec.ts (hunk 1 above)
Comment (webmaster128) [reply packages/bcp/src/cosmwasmcodec.spec.ts:4]: God point, I forgot that, will update

**Pull request merged**